### PR TITLE
Generating Code with stable version 2024-04

### PR DESCRIPTION
**Pull request by achauhan-scc**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/machinelearningservices/Azure.ResourceManager.MachineLearning/src/Generated/Models/PublicNetworkAccess.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+#nullable disable
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.ResourceManager.MachineLearning.Models
+{
+    /// <summary> Whether requests from Public Network are allowed. </summary>
+    public readonly partial struct PublicNetworkAccess : IEquatable<PublicNetworkAccess>
```
Comment (archerzz) [inline sdk/machinelearningservices/Azure.ResourceManager.MachineLearning/src/Generated/Models/PublicNetworkAccess.cs:14]: This enum is actually the same as https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/machinelearningservices/Azure.ResourceManager.MachineLearning/src/Generated/Models/MachineLearningPublicNetworkAccessType.cs

And in referred places like `MachineLearningWorkspacePatch`, the type of `PublicNetworkAccessType` property is changed from `MachineLearningPublicNetworkAccessType` to `PublicNetworkAccess`.

Is that by accident?

File: sdk/machinelearningservices/Azure.ResourceManager.MachineLearning/src/Generated/Models/EndpointServiceConnectionStatus.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+#nullable disable
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.ResourceManager.MachineLearning.Models
+{
+    /// <summary> Connection status of the service consumer with the service provider. </summary>
+    public readonly partial struct EndpointServiceConnectionStatus : IEquatable<EndpointServiceConnectionStatus>
```
Comment (archerzz) [inline sdk/machinelearningservices/Azure.ResourceManager.MachineLearning/src/Generated/Models/EndpointServiceConnectionStatus.cs:14]: This enum is different from `MachineLearningPrivateEndpointServiceConnectionStatus` in that it doesn't have `Timeout` value. Just want to confirm that it's expected.

**Review by archerzz — COMMENTED** [review]
Generally good. Just to confirm that two changes in enum types are expected.

File: sdk/machinelearningservices/Azure.ResourceManager.MachineLearning/src/Generated/Models/PublicNetworkAccess.cs (hunk 1 above)
Comment (achauhan-scc) [reply sdk/machinelearningservices/Azure.ResourceManager.MachineLearning/src/Generated/Models/PublicNetworkAccess.cs:14]: MachineLearningPublicNetworkAccessType is part of OnlineEndpoint defined in https://github.com/Azure/azure-rest-api-specs/blob/main/specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2024-04-01/mfe.json

PublicNetworkAccess is part of workspace patch "WorkspacePropertiesUpdateParameters" defined in https://github.com/Azure/azure-rest-api-specs/blob/main/specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2024-04-01/machineLearningServices.json

File: sdk/machinelearningservices/Azure.ResourceManager.MachineLearning/src/Generated/Models/EndpointServiceConnectionStatus.cs (hunk 2 above)
Comment (achauhan-scc) [reply sdk/machinelearningservices/Azure.ResourceManager.MachineLearning/src/Generated/Models/EndpointServiceConnectionStatus.cs:14]: MachineLearningPrivateEndpointServiceConnectionStatus is part of SharedPrivateLinkResourceProperty https://github.com/Azure/azure-rest-api-specs/blob/main/specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2024-04-01/machineLearningServices.json

EndpointServiceConnectionStatus is part of RegistryPrivateLinkServiceConnectionState https://github.com/Azure/azure-rest-api-specs/blob/main/specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2024-04-01/registries.json

**Review by archerzz — APPROVED** [review]

**Review by ArthurMa1978 — APPROVED** [review]

**Pull request merged**